.github/workflows/tests.yml

```
@@ -21,129 +21,77 @@ jobs:
-      - name: Build
-        run: dotnet build ./src/Playwright.Tests/Playwright.Tests.csproj -c Debug -f netcoreapp3.1
-      - name: Download Browsers
-        run: ./src/Playwright/bin/Debug/netstandard2.0/playwright.ps1 install
-      - name: Run tests (!Linux)
-        if: ${{ matrix.os != 'ubuntu-latest' }}
-        env:
-          BROWSER: chromium
+          dotnet-version: 5.0.x
+      - name: Install prerequisites and download drivers
+        shell: pwsh
+        run: .\build.ps1 driver
+      - name: Building...
+        shell: pwsh
         run: |
-          dotnet test ./src/Playwright.Tests/Playwright.Tests.csproj -c Debug -f netcoreapp3.1 --logger "trx;LogFileName=TestResults.xml" -- NUnit.NumberOfTestWorkers=1
-      - name: Run tests (Linux)
-        if: ${{ matrix.os == 'ubuntu-latest' }}
-        env:
-          BROWSER: chromium
+          dotnet build -f netcoreapp3.1 .\Playwright\Playwright.csproj
+          dotnet build -f netcoreapp3.1 .\Playwright.Tests\Playwright.Tests.csproj
+        working-directory: src
+      - name: Installing Browsers and dependencies...
+        shell: pwsh
         run: |
-          xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- bash -c "ulimit -c unlimited && dotnet test ./src/Playwright.Tests/Playwright.Tests.csproj -c Debug -f netcoreapp3.1 --logger \"trx\" -- NUnit.NumberOfTestWorkers=1"
-  test_cli:
-    name: CLI and NUnit
-    runs-on: ${{ matrix.os }}
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [windows-latest, ubuntu-latest, macos-latest]
-    steps:
-      - uses: actions/checkout@v2
-      - uses: microsoft/playwright-github-action@v1
-      - name: Setup .NET Core (On Mac)
-        if: ${{ matrix.os == 'macos-latest' }}
-        uses: actions/setup-dotnet@v1
-        with:
-          dotnet-version: 5.0.100
+          dotnet run -p .\src\Playwright\Playwright.csproj -f net5.0 -- install
+          dotnet run -p .\src\Playwright\Playwright.csproj -f net5.0 -- install-deps
+      - name: Running tests... 
+        shell: bash
         env:
-          DOTNET_INSTALL_DIR: /users/runner/alternate/dotnet
-      - name: Setup .NET Core (Not on mac)
-        if: ${{ matrix.os != 'macos-latest' }}
-        uses: actions/setup-dotnet@v1
-        with:
-          dotnet-version: 5.0.100
-      - name: Download Drivers
-        run: dotnet run -p ./src/tools/Playwright.Tooling/Playwright.Tooling.csproj -- download-drivers --basepath .
-      - name: Build CLI
-        run: dotnet build ./src/Playwright.CLI/
-      - name: Build NUnit
-        run: dotnet build ./src/Playwright.NUnit/
+          BROWSER: CHROMIUM
+        working-directory: src
+        run: |
+          xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- bash -c "ulimit -c unlimited && dotnet test ./Playwright.Tests/Playwright.Tests.csproj -c Debug -f net5.0 --logger \"trx\" -- NUnit.NumberOfTestWorkers=1"
```

```suggestion
          xvfb-run dotnet test ./Playwright.Tests/Playwright.Tests.csproj -c Debug -f net5.0 --logger "trx" -- NUnit.NumberOfTestWorkers=1
```
lets do what our users do, for Python we also have it like that.

---

Sure, this is a legacy thing that was in here since the beginning. If we can simplify it, I'm all for it.